src/crowdin_client.cpp

```diff
@@ -196,20 +196,15 @@ dispatch::future<CrowdinClient::UserInfo> CrowdinClient::GetUserInfo()
             const json& d = r["data"];
             UserInfo u;
             u.login = str::to_wstring(d["username"]);
-            u.name = str::to_wstring(d.value("fullName", ""));
-            if(u.name.empty())
+            try { u.name = str::to_wstring(d.at("fullName")); }
+            catch(...)
             {
-                // Take care that both first and last name can be empty
-                auto name = d["firstName"];
-                if(name.is_string())
-                    u.name += str::to_wstring(name);
-                if(!u.name.empty())
-                    u.name += ' ';
-                name = d["lastName"];
-                if(name.is_string())
-                    u.name += str::to_wstring(name);
+                std::string firstName, lastName;
+                try { firstName = d.at("firstName"); } catch(...) {}
+                try { lastName = d.at("lastName"); } catch(...) {}
+                u.name = str::to_wstring(firstName + ' ' + lastName);
             }
-            if(u.name.empty())
+            if(u.name.empty() || u.name == L" ")
```

Comment:
Refactored and commented d81778a21f556f1e0eaa2691f3fb65fe00d3c446 what hopefully will make it more clear an self-explanatory for you and anybody else.

> How does this "fix GetUserInfo()" and why is it done?

Because earlier implementation might crash if e.g. `firstName` or `lastName` are either `null` or  missing (what is the one of above cases of `First` or `Last` name field is set/left empty in Crowdin Enterprise account settings. And above happened for me at least once.

> value("key", "default") is appropriate here

It raises exception when there is e.g. `"key": null` in JSON response what happened for me too as `"firstName": null"` or `"fullName": null` - already don't remember which exactly but due of such sometime undefined/unforeseen response format (might be due v2 API are still in Beta or so) - I decided to solve it with `try-catch` drastically :) (i.e. to make `GetUserInfo()` fully tolerant to **any** surprises with JSON response format/structure already)

> silently eating catch-all less so

But what error/message or even log entry are you expect here if most of catches just mean that field is empty or so but not actual error? While to figure out whether it's some unexpected response from server or something in our handling logic - is complicated here sometime or at least not worthing efforts since does not add any value for customer.  This again due of not absolutely clearly defined above mentioned variations of response from the server - i.e. when some of above keys are missing and when `null` what still potentially might change during v2 API beta I guess. Though if you prefere still some log entry there (just like "Missing First name") - please let me know so I will add them.